src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj

```diff
@@ -78,4 +78,23 @@
     </ProjectReference>
   </ItemGroup>
 
+  <Target Name="GenerateBuildMetadata" BeforeTargets="BeforeBuild">
+    <WriteLinesToFile
+      File="$(IntermediateOutputPath)Generated_BuildMetadata.cs"
+      Lines="// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+internal static class BuildMetadata
+{
+    public const string NetCoreAppCurrentVersion = &quot;$(NetCoreAppCurrentVersion)&quot;%3B
```

Comment:
Why do we need this one? I would keep it out if not necessary.

Reply:
Just a showcase I guess 😅
i.e.  it's possible to extend it and make it configurable etc. e.g. the requesting project can set RequiredMetadata="Foo;Bar;Baz" and it would generate those strings and put the rest behind `#if 0` (so it doesn't cost anything).